apps/server/server.cpp

```diff
@@ -511,6 +505,155 @@ void invoke_sdk_api(payload::ClientRequest buff_recv) {
+        uint8_t *buffer = new uint8_t[length];
+        aditof::Status status = eeprom->read(address, buffer, length);
+        if (status == aditof::Status::OK) {
+            buff_send.add_bytes_payload(buffer, length);
+        }
+        delete[] buffer;
+        buff_send.set_status(static_cast<::payload::Status>(status));
+        break;
+    }
+
+    case EEPROM_WRITE: {
+#ifdef DEBUG
+        cout << "Eepromrite function\n";
+#endif
+        // Check if client has sent exactly one name to identify the EEPROM
+        if (buff_recv.device_data().eeproms_size() != 1) {
+            buff_send.set_message(
+                "Failed to idenify EEPROM. Exactly one name for the eeprom is "
+                "accepted. Number of names provided: " +
+                std::to_string(buff_recv.device_data().eeproms_size()));
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
+            break;
+        }
+        std::string eName = buff_recv.device_data().eeproms(0).driver_name();
+
+        // We lookup the EEPROM instance by name
+        auto eeprom = findEepromByName(eName);
+        if (!eeprom) {
+            buff_send.set_message("Failed to idenify EEPROM with name: " +
+                                  eName);
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
+        }
+
+        uint32_t address = static_cast<uint32_t>(buff_recv.func_int32_param(0));
+        size_t length = static_cast<size_t>(buff_recv.func_int32_param(1));
+        const uint8_t *buffer = reinterpret_cast<const uint8_t *>(
+            buff_recv.func_bytes_param(0).c_str());
+
+        aditof::Status status = eeprom->write(address, buffer, length);
+        buff_send.set_status(static_cast<::payload::Status>(status));
+        break;
+    }
+
+    case EEPROM_CLOSE: {
+#ifdef DEBUG
+        cout << "EepromClose function\n";
+#endif
+        // Check if client has sent exactly one name to identify the EEPROM
+        if (buff_recv.device_data().eeproms_size() != 1) {
+            buff_send.set_message(
+                "Failed to idenify EEPROM. Exactly one name for the eeprom is "
+                "accepted. Number of names provided: " +
+                std::to_string(buff_recv.device_data().eeproms_size()));
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
+            break;
+        }
+        std::string eName = buff_recv.device_data().eeproms(0).driver_name();
+
+        // We lookup the EEPROM instance by name
+        auto eeprom = findEepromByName(eName);
+        if (!eeprom) {
+            buff_send.set_message("Failed to idenify EEPROM with name: " +
+                                  eName);
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
+        }
+        break;
```

Comment:
shouldn't this `break` be in the `if (!eeprom)` body? It always breaks and this line will never be reached ` 647: aditof::Status status = eeprom->close();`. Is this some temporary desired behaviour?

Reply:
yes, it should be inside the body. It's fixed now.